lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```diff
@@ -73,70 +77,89 @@ public Integer call() throws Exception {
+    private final FutureTask<Integer> task;
+    private final ExecutorService executor;
+
+    AsyncInitiator(final FutureTask<Integer> task, final ExecutorService executor) {
+      this.task = task;
+      this.executor = executor;
+    }
+
+    public Boolean call() {
+      executor.execute(task);
+      return true;
+    }
+  }
+
   /**
    * Asynchronously increment the input parameter.
    * @param input An integer object whose value is to be incremented by one.
    * @return The input parameter incremented by one or zero for a timeout.
    * @throws InterruptedException Thread was interrupted by another thread.
    * @throws ExecutionException An exception was thrown an internal processing function.
+   * @throws InvalidIdentifierException The call identifier is invalid.
+   * @throws Exception The asynchronous processing generated an exception.
    */
-  public int apiCall(final Integer input) throws InterruptedException, ExecutionException {
+  public int apiCall(final Integer input) throws Exception {
     // Create a future to run the asynchronous processing.
     final long identifier = idCounter.getAndIncrement();
     final FutureTask<Integer> task =
         new FutureTask<>(new AsynchronousIncrementer(input, identifier, incrementerSleepTimeMillis, blocker));
     taskQueue.add(task);
 
     LOG.log(Level.INFO, "Running the incrementer...");
-    // Start the task and block the caller until it completes.
-    executor.execute(task);
-    if (blocker.block(identifier)) {
+    if (blocker.block(identifier, new AsyncInitiator(task, executor))) {
       LOG.log(Level.INFO, "Call timed out...");
       // Timeout occurred before the asynchronous processing completed.
       return 0;
     }
     int result = -1;
     LOG.log(Level.INFO, "Call getting task result...");
-    result = task.get();
-    return result;
+    return task.get();
   }
 
   /**
-   * Insure all test tasks have completed.
+   * Ensure all test tasks have completed.
+   * @throws ExecutionException Asynchronous processing generated an exception.
    */
-  public void complete() throws ExecutionException {
-    try {
-      for (final FutureTask<Integer> task : taskQueue) {
+  public void close() throws ExecutionException {
+    for (final FutureTask<Integer> task : taskQueue) {
+      try {
         task.get();
+      } catch (final ExecutionException ee) {
+        // When asynchronous processing completes after the call to MultiAsyncToSync.block() times
+        // out, the call to MultiAsyncToSync.release() will throw this an InvalidIdentifierException
+        // and it is expected; otherwise, we flag the error.
+        if (!(expectTimeout && ee.getCause() instanceof InvalidIdentifierException)) {
+          LOG.log(Level.INFO, "Caught exception waiting for completion...", ee);
+        }
+      } catch (final Exception e) {
+        LOG.log(Level.INFO, "Caught exception waiting for completion...", e);
```

Comment:
I would just have
```java
catch (final Exception e) {
  LOG.log(Level.WARNING, "Caught exception waiting for completion...", e);
}
```
it will log the stack trace not only of the immediate exception, but of the subsequent exceptions in the chain.